var/spack/repos/builtin/packages/eccodes/package.py

```diff
@@ -70,7 +72,16 @@ class Eccodes(CMakePackage):
     )
     variant("png", default=False, description="Enable PNG support for decoding/encoding")
     variant(
-        "aec", default=False, description="Enable Adaptive Entropy Coding for decoding/encoding"
+        "aec",
+        default=False,
+        when="@:2.24",
+        description="Enable Adaptive Entropy Coding for decoding/encoding",
+    )
+    variant(
+        "aec",
+        default=True,
+        when="@2.25:",
+        description="Enable Adaptive Entropy Coding for decoding/encoding",
```

Comment:
```suggestion
        "aec", default=True, description="Enable Adaptive Entropy Coding for decoding/encoding"
```

Reply:
This PR's implementation reflects the defaults of aec in different versions of eccodes. This means users get the same default value for aec, no matter if they install it with or without spack.
This suggestion could surprise users who are already familiar with older versions of eccodes and their defaults.

Reply:
> [The only issue is that this currently does not work](https://github.com/spack/spack/pull/40770#discussion_r1378086193)

And although I respect the `if they install it with or without spack` argument, I don't consider it strong enough for having (currently) redundant (if not confusing) code.

The users surprise us all the time. Time for a little revenge.

Reply:
Ok agreed the code is at least simplified with this suggestion.